data_structures/binary_search_tree_OOP_approach.cpp

```diff
@@ -0,0 +1,205 @@
+/**
+ * \file
+ * \brief A simple tree implementation using structured nodes
+ * \Object Oriented approach
+ */
```

Comment:
As @kvedala mentioned, please update this comment block like in [`skip_list`](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/data_structures/skip_list.cpp#L1-#L14).